apps/docs/app/ui/components/componentExample/ComponentExample.tsx

```diff
@@ -0,0 +1,101 @@
+"use client";
+
+import { useState, memo, useCallback, useMemo, type ReactNode } from "react";
+import clsx from "clsx";
+
+import { CodeIcon } from "@/components/icon";
+import { ToggleButton } from "@/components/toggleButton/ToggleButton.tsx";
+
+import ComponentPreviewWrapper from "./ComponentPreviewWrapper.tsx";
+
+import "./componentExample.css";
+
+interface CommonProps {
+    className?: string;
+    isOpen?: boolean;
+}
+
+interface CodeProps extends CommonProps {
+    type: "code";
+    code: ReactNode;
+}
+
+interface PreviewProps extends CommonProps {
+    type: "preview";
+    preview: ReactNode;
+}
+
+interface BothProps extends CommonProps {
+    type: "both";
+    code: ReactNode;
+    preview: ReactNode;
+}
+
+export type ComponentExampleProps = CodeProps | PreviewProps | BothProps;
+
+const ComponentExample = memo(({
+    type = "both",
+    className,
+    isOpen = false,
+    ...props
+}: ComponentExampleProps) => {
+    const [showCode, setShowCode] = useState(isOpen);
+
+    const showBothComponent = useMemo(() => type === "both", [type]);
```

Comment:
all 3 show properties aren't complex calculation. I feel like useMemo probably slows the rendering instead of helping. Was there a specific need for this?